Framework/Kernel/inc/MantidKernel/CompositeValidator.h

```diff
@@ -65,12 +70,22 @@ class DLLExport CompositeValidator : public IValidator {
   }
 
 private:
-  /// Private Copy constructor: NO DIRECT COPY ALLOWED
-  CompositeValidator(const CompositeValidator &);
   /// Verify the value with the child validators
   std::string check(const boost::any &value) const override;
+  /// Verify the value with the child validators with logical "and"
+  std::string checkLogicalAnd(const boost::any &value) const;
+  /// Verify the value with the child validators with logical "and"
+  std::string checkLogicalOr(const boost::any &value) const;
```

Comment:
Maybe that is clearer yes. The comment is also wrong here.

I originally implemented it as a separate class, but then we can just rename the old one as that would break backwards compatibility with the python api which exposes `CompositeValidator`.